#+HTML_HEAD: <style> pre.src { white-space: pre-wrap; } </style>
#+PANDOC_OPTIONS: self-contained:t


This is the repository for reproducing results of Human Embryonic RNA Editome.
 
* List of all important files from Zenodo

Download files from Zenodo (DOI: 10.5281/zenodo.5804921) at the corresponding location and uncompress them as specified.


** ~pipeline.validation.bcf.files.tar.gz~

- Needed by Figure 1D.
- Download the archive as ~./zenodo-archives/pipeline.validation.bcf.files.tar.gz~.
- Uncompress the archive by running ~tar -xzvf ./zenodo-archives/pipeline.validation.bcf.files.tar.gz~.


** ~editome.files.tar.gz~
   
- Contains the identified editome (i.e., all edits identified in each sample): ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.dt.txt.gz~. See below for its column description.
- Other important files included:
  - All variants (including edits and other non-edit variants) identified in each sample: ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.with.event.summary.dt.txt.gz~
  - BED file for all identified edits (editing position only): ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.bed~
- Needed by Figure 2A,2B,2C,2D,2E,2F,5A,5B,5C,5E; Supplementary Table 3,4; Supplementary Figure 2A,2B,3A,3B.
- Download the archive as ~./zenodo-archives/editome.files.tar.gz~.
- Uncompress the archive by running ~tar -xzvf ./zenodo-archives/editome.files.tar.gz~.

*** Columns for edits

 | Column                 | Meaning                                                                                                                                                |
 |------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------|
 | ID                     | ID of the editing site                                                                                                                                 |
 | SAMPLE                 | Sample ID (as GSM acccession)                                                                                                                          |
 | SUBSET                 | Subset of this editing site. Alu: on Alu elements; RepNOTAlu: on repetitive elements that are not Alu; nonRep: not on repetitive elements              |
 | AC                     | Number of A-to-G mismatch reads on this site reported by GATK                                                                                          |
 | AN                     | Number of reads on this site reported by GATK                                                                                                          |
 | AF                     | Editing level; equals AC/AN; reported by GATK                                                                                                          |
 | gse                    | GSE accession of the dataset this sample comes from                                                                                                    |
 | stage                  | Stage of this sample                                                                                                                                   |
 | is.normal              | TRUE if this sample is normal, and FALSE otherwise                                                                                                     |
 | disease                | Description of disease for this sample                                                                                                                 |
 | treatment              | Description of treatment for this sample                                                                                                               |
 | maternal.age           | Description of maternal age for this sample                                                                                                            |
 | developmental.day      | Description of developmental day for this sample                                                                                                       |
 | cell.line              | Description of cell line for this sample (only meaningful to hESC samples)                                                                             |
 | srr.count              | Number of SRR runs from this sample                                                                                                                    |
 | srr.mean.avgspotlen    | Mean AvgSpotLen of SRR runs for this sample                                                                                                            |
 | srr.total.bytes        | Total bytes of SRR runs for this sample                                                                                                                |
 | srr.total.bases        | Total bases of SRR runs for this sample                                                                                                                |
 | srr.total.avgreadcount | Total AvgReadCount of SRR runs for this sample                                                                                                         |
 | site.occurrence        | Site occurrence of this editing site in those of all 2,071 samples with matched ~stage~ and ~is.normal~                                                |
 | CHROM                  | Chromosome of this editing site                                                                                                                        |
 | POS                    | Position of this editing site (1-based)                                                                                                                |
 | REF                    | Reference allele for this editing site (based on genomic Watson strand)                                                                                |
 | ALT                    | Alternative allele for this editing site  (based on genomic Watson strand)                                                                             |
 | event.summary          | Editing event summary for this site. Note that this could be either 'A>G' or 'A>G;T>C' (when two transcripts of opposite direction overlap this site). |


 

** ~RE.files.tar.gz~

- Contains the following (see below for column description):
  - The identified REs in normal samples: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.dt.txt.gz~ 
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~
- Other important files included:
  - Edits in normal samples: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.dt.txt.gz~ 
  - Occurrence of RE-matching edits in each normal sample: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.site.recurrence.comparison.CJ.dt.txt.gz~
  - Observed edits identified in each normal sample, on valid genes only: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.observed.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~
  - SnpEff annotation for observed edits identified in normal samples, on valid genes only: ~./result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/snpEff.annotation.for.subset.observed.edits.dt.txt.gz~
  - Edit recurrence in each GSE133854 sample: ~./result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/GSE133854.all/subset.site.recurrence.comparison.CJ.dt.txt.gz~
- Needed by Figure 3B,3C,3D,3E,4B,4C,5A,5B,5C,5E,6B,6C,6D; Supplementary Table 3,4,5,6,7; Supplementary Figure 1,2A,2B,3A,3B,4.
- Download the archive as ~./zenodo-archives/RE.files.tar.gz~.
- Uncompress the archive by running ~tar -xzvf ./zenodo-archives/RE.files.tar.gz~.

*** Columns for REs  

 This table inherits columns from the editome in ~editome.files.tar.gz~ above, plus the following columns:


 | Column                                   | Meaning                                                                                                                                                                                                                   |
 |------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | group                                    | Group of this sample (named with ~stage~ @ ~is.normal~)                                                                                                                                                                   |
 | depth                                    | Read coverage deduced from early bam alignment (~alignment.sorted.withRG.dedup.converted.bq.sorted.without.splicing.junction.SN.bam~; see ~rule S52_1__check_variant_converage_of_merged_bam~ in ~pipeline.v3.part3.smk~) |
 | total.sample.count.for.this.sample.group | (deprecated)                                                                                                                                                                                                              |
 | total.sample.count                       | Total sample count for this group                                                                                                                                                                                         |
 | site.occurrence.for.this.group           | Site occurrence of this editing site across all samples in this group (identical to ~site.occurrence~ for ~all.normal.samples~)                                                                                           |
 | group.occurrence.pct                     | site.occurrence.for.this.group / total.sample.count.for.this.sample.group                                                                                                                                                 |


*** Columns for snpEff annotation

    
 | Column               | Meaning                                                                                                                                                                                                                                       |
 |----------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | Gene_Name            | Name of gene                                                                                                                                                                                                                                  |
 | Gene_ID              | Ensembl ID of Gene                                                                                                                                                                                                                            |
 | Annotation           | SnpEff Annotation for this edit                                                                                                                                                                                                               |
 | Annotation.pasted    | pasted Annotation for the site on the given gene locus (it is possible that the annotation of this site on different transcripts of the same gene locus might differ)                                                                         |
 | Annotation.corrected | corrected Annotation by Annotation priority (see Methods)                                                                                                                                                                                     |
 | Annotation.class     | "exonic.or.splicing.related", i.e., the edit is annotated as exonic or related to splicing (referred as 'exonic' in manuscript' , or "purely.intronic", i.e., the edit is annotated as purely intronic (referred as 'intronic' in manuscript) |




** ~microRNA.files.tar.gz~

- Contains the following:
  - TargetScan output for unedited transcripts: ~result/A02_8__get_editing_effect_on_miRNA_binding_sites/step09__concatenate_TargetScan_results_across_all_chromosomes/32/gencode.3utr.all.chromosomes.concatenated.headless.TargetScan.output.gz~.
  - TargetScan output for edited transcripts: ~result/A02_8__get_editing_effect_on_miRNA_binding_sites/step10__concatenate_edited_TargetScan_results_across_all_chromosomes/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/32/edited.gencode.3utr.all.chromosomes.but.chrY.concatenated.headless.TargetScan.output.gz~
- Needed by Figure 6B,6C,6D; Supplementary Figure 4.
- Download the archive as ~./zenodo-archives/microRNA.files.tar.gz~.
- Uncompress the archive by running ~tar -xzvf ./zenodo-archives/microRNA.files.tar.gz~.


* Prerequisites

** Tools

*** Basic tools

- Java
- Perl
- pigz (to speed up gzip compression)
- Python 3
- R
- Snakemake >= 5.10.0 (to support the ~allow_missing~ argument of ~expand~) 
  
*** Bioinformatics command-line tools

- Trim Galore! == 0.6.6
- fastp
- BWA
- Picard
- Samtools
- Galaxy tools (for subsetting MAFs)
  - The core scripts used here are already included in this repository:
    - ~tools/galaxy/interval_maf_to_merged_fasta.py~
    - ~tools/galaxy/galaxy/tools/util/maf_utilities.py~
  - If you need to retrieve these python scripts by yourself, delete these two scripts above and run the following bash commands:
#+BEGIN_SRC bash
git clone https://github.com/galaxyproject/tools-iuc.git tools/galaxy-tools-iuc
git clone -b release_21.01 git@github.com:galaxyproject/galaxy.git tools/galaxy.21.01
mkdir -p tools/galaxy/galaxy/tools/util/

cp tools/galaxy-tools-iuc/tools/genebed_maf_to_fasta/interval_maf_to_merged_fasta.py tools/galaxy/interval_maf_to_merged_fasta.py
cp -L tools/galaxy.21.01/lib/galaxy/tools/util/maf_utilities.py tools/galaxy/galaxy/tools/util/maf_utilities.py
#+END_SRC
- GATK == 3.6.0
  - Needs to be put as ~tools/GATK-3.6.0/GenomeAnalysisTK.jar~
- BCFtools
- BLAT
- SnpEff
- TargetScan v7.0
  - Downloaded from [[http://www.targetscan.org/vert_80/vert_80_data_download/targetscan_70.zip]]
  - Needs to be put as ~tools/targetscan_70/targetscan_70.pl~
- The following UCSC tools (can be installed via bioconda)
  - genePredToBed (https://anaconda.org/bioconda/ucsc-genepredtobed)
  - gtfToGenePred (https://anaconda.org/bioconda/ucsc-gtftogenepred)
- The ~tools/convertCoordinates_classpath/convertCoordinates.java~ kindly provided by Dr. Gokul Ramaswami needs to be compiled before use. Run ~javac tools/convertCoordinates_classpath/convertCoordinates.java~ to get the compiled ~tools/convertCoordinates_classpath/convertCoordinates.class~.
- Two Sample Logos (tsl; [[http://www.twosamplelogo.org]])
  - We packed a working tsl at ~tools/tsl~ in this repository. If you want to download the tsl again, the tsl needs to be put as ~tools/tsl~ such that the command-line is at ~tools/tsl/cgi-bin/tsl~
  - Note that this tool depends on Ruby.


*** Python packages

- pandas
- biopython (whose script ~maf_build_index.py~ will be used)

*** R packages


**** CRAN packages

- data.table
- foreach
- ggalluvial
- ggpubr
- ggtext
- iterators
- magick
- magrittr
- R.utils
- rmarkdown
- scales
- stringr

**** Bioconductor packages

- clusterProfiler
- GEOmetadb
- org.Hs.eg.db == 3.12.0





** Datasets

*** Reference genome (hg38) from UCSC

#+BEGIN_SRC bash
mkdir -p external/contigs/
wget -P external/contigs/ http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz
zcat external/contigs/hg38.fa.gz > external/contigs/hg38.fa
samtools faidx external/contigs/hg38.fa
#+END_SRC

*** GENCODE GTF annotation (human, version 32)

#+BEGIN_SRC bash
mkdir -p external/reference.gene.annotation/GENCODE.annotation/32/
wget -P external/reference.gene.annotation/GENCODE.annotation/32/  ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.annotation.gtf.gz
zcat external/reference.gene.annotation/GENCODE.annotation/32/gencode.v32.annotation.gtf.gz > external/reference.gene.annotation/GENCODE.annotation/32/gencode.v32.annotation.gtf
#+END_SRC

*** GENCODE transcripts fasta file (human, version 32)

#+BEGIN_SRC bash
mkdir -p external/contigs/
wget -P external/contigs/ ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.transcripts.fa.gz
#+END_SRC


*** dbSNP version 151

#+BEGIN_SRC bash
mkdir -p external/dbSNP.vcf/151/common_all/
wget -P external/dbSNP.vcf/151/common_all/  https://ftp.ncbi.nlm.nih.gov/snp/organisms/human_9606/VCF/00-common_all.vcf.gz
wget -P external/dbSNP.vcf/151/common_all/  https://ftp.ncbi.nlm.nih.gov/snp/organisms/human_9606/VCF/00-common_all.vcf.gz.tbi
ln -s external/dbSNP.vcf/151/common_all/00-common_all.vcf.gz external/dbSNP.vcf/151/common_all/dbSNP.vcf.gz
ln -s external/dbSNP.vcf/151/common_all/00-common_all.vcf.gz.tbi external/dbSNP.vcf/151/common_all/dbSNP.vcf.gz.tbi
#+END_SRC


*** UCSC MAF tracks

Note: 
1. The MAF files must be decompressed before use, taking up to 153G in total.
2. You need to build index before use.

#+BEGIN_SRC bash
mkdir -p ./external/UCSC.maf30way/
for chr in `seq -f "chr%g" 1 22 ` chrX chrY
do
  wget -P ./external/UCSC.maf30way/ https://hgdownload.soe.ucsc.edu/goldenPath/hg38/multiz30way/maf/${chr}.maf.gz
  zcat ./external/UCSC.maf30way/${chr}.maf.gz > ./external/UCSC.maf30way/${chr}.maf
done

## build index

for chr in `seq -f "chr%g" 1 22 ` chrX chrY
do
    maf_build_index.py ./external/UCSC.maf30way/${chr}.maf
done

#+END_SRC

*** UCSC tracks (others)

Prepare the UCSC track files using Table Browser ([[https://www.genome.ucsc.edu/cgi-bin/hgTables/]]) as described below. All tracks should be from the hg38 assembly.

| Dataset                        | group                      | track              | filter                             | output format | rename as                                                                           |
|--------------------------------+----------------------------+--------------------+------------------------------------+---------------+-------------------------------------------------------------------------------------|
| knownGene (GENCODE version 32) | Genes and Gene Predictions | GENCODE v32        | (none)                             | (default)     | ~external/UCSC.Table.Browser.knownGene.GENCODE/32/knownGene~                        |
| dbSNP cDNA-flagged             | Variation                  | Flagged SNPs (151) | molType does match cDNA            | BED           | ~external/UCSC.Table.Browser.dbSNP/151/flagged.cDNA.only/dbSNP.bed~                 |
| Alu repeats                    | Repeats                    | RepeatMasker       | repFamily does match Alu           | BED           | ~external/UCSC.Table.Browser.repeatmasker/repFamily.Alu/repeatmasker.bed~           |
| Simple repeats                 | Repeats                    | RepeatMasker       | repFamily does match Simple_repeat | BED           | ~external/UCSC.Table.Browser.repeatmasker/repFamily.Simple_repeat/repeatmasker.bed~ |
| Non-Alu repeats                | Repeats                    | RepeatMasker       | repFamily does match Alu           | BED           | ~external/UCSC.Table.Browser.repeatmasker/repFamily.Alu/repeatmasker.bed~           |


*** Genomic VCF files from worldwide cohort studies

Download the corresponding *vcf.gz files (and their .tbi indices) as described below, and rename each individual chromosome-level VCF file as ~external/outer_vcf/{OUTER_VCF_NAME}/{OUTER_VCF_SUBSET}/outer.VCF~ (and its index as ~external/outer_vcf/{OUTER_VCF_NAME}/{OUTER_VCF_SUBSET}/outer.VCF.tbi~) where ~{OUTER_VCF_NAME}~ is described below for each study and ~{OUTER_VCF_SUBSET}~ is ~chr1, chr2, ..., chrX, chrY~:

| Study                          | URL for official site                                                                                                     | ~OUTER_VCF_NAME~              |
|--------------------------------+---------------------------------------------------------------------------------------------------------------------------+-------------------------------|
| UWashington EVS                | [[https://evs.gs.washington.edu/EVS/]]                                                                                        | ~UWashington.EVS~             |
| NCBI ALFA (version 2020.03.04) | [[https://ftp.ncbi.nih.gov/snp/population_frequency/archive/release_1/]]                                                      | ~NCBI.ALFA.2020.03.04~        |
| gnomAD (v2.1.1, exomes)        | [[https://gnomad.broadinstitute.org]]                                                                                         | ~gnomAD_v2.1.1_exomes~        |
| gnomAD (v2.1.1, genomes)       | [[https://gnomad.broadinstitute.org]]                                                                                         | ~gnomAD_v2.1.1_genomes~       |
| gnomAD (v3.0, genomes)         | [[https://gnomad.broadinstitute.org]]                                                                                         | ~gnomAD_v3.0_genomes~         |
| 1000Genomes                    | [[http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/release/20190312_biallelic_SNV_and_INDEL/]] | ~1000Genomes.phased.genotype~ |

*** GEOmetadb sqlite

Use the R Bioconductor package ~GEOmetadb~ to download the ~GEOmetadb.sqlite.gz~, uncompress it, and rename it as ~external/NCBI.GEOmetadb/GEOmetadb.sqlite~.

*** REDIPortal 2021 hg38 annotation

#+BEGIN_SRC bash
mkdir -p external/REDIPortal/hg38/
wget -P external/REDIPortal/hg38/ http://srv00.recas.ba.infn.it/webshare/ATLAS/donwload/TABLE1_hg38.txt.gz
#+END_SRC


*** TargetScan miRNA family
#+BEGIN_SRC bash
mkdir -p external/TargetScan/
wget -P external/TargetScan/ http://www.targetscan.org/vert_80/vert_80_data_download/miR_Family_Info.txt.zip
unzip -d external/TargetScan/ ./external/TargetScan/miR_Family_Info.txt.zip

awk 'OFS="\t"{if($6==2 && $3==9606)print $1,$2,$3}' ./external/TargetScan/miR_Family_Info.txt | sort | uniq > ./external/TargetScan/miR_Family_Info.human.temp.txt
python tools/get_miR_family.py ./external/TargetScan/miR_Family_Info.human.temp.txt ./external/TargetScan/miR_Family_Info.txt ./external/TargetScan/miR_Family_Info.human.txt

#+END_SRC

* Notes before running codes
** About the codes themselves

- All codes are Linux Bash Shell commands.
- *WARNING*: 
  - Due to the large sample size, all ~snakemake~ commands before producing figures take a vast amount of cores and memory. The users are strongly recommended to adjust the ~thread_*~ and ~default_Xmx~ parameters and run these on a cluster.
- *NOTES*:
  - The ~snakemake~ commands below are ended with a ~-n~ (dry-run). Running with ~-n~ will only list all the tasks planned to run (plus their dependencies) and will not really run/submit these tasks. The user can remove the ~-n~ parameter and run the commands again once agreeing with the plan.


** About ~--job~ in snakemake commands

For each ~snakemake~ command, the ~--jobs~ parameter (i.e., number of cores to use in local mode and number of concurrent jobs allowed to run in cluster mode) is restricted to 1 here for demonstration only.

** For HPC users

If you'd like to run snakemake commands on clusters, please add the ~--cluster {your-own-cluster-submission-command}~ option to make snakemake run on clusters. See [[https://snakemake.readthedocs.io/en/stable/executing/cluster.html]] for more details.

* Identify the A-to-I RNA editome


** 1. Construct splice-aware genomes

- This step constructs splice-aware genomes for the subsequent RNA editing calling and expression profiling.

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.smk --config threads_indexing=36 threads_trimming=1 threads_aligning=36 threads_merging_bams=1 threads_calling_variants=36 threads_auxiliary_processing=1 threads_auxiliary_processing_parallel=6 --jobs 1 -prk --nolock \
    result/s05_1__index_contig_with_annotation/hg38.fa/32/bwa-index-10.1038_nmeth.2330/{95,96,75,103,120,144,145,45,85}/finished -n
#+END_SRC

** 2. Deploy samples

- 18 datasets / 2071 samples in total

| No. | Dataset   | # samples used | ~{DATASET_NAME}~ (used by later ~snakemake~ commands) |
|-----+-----------+----------------+------------------------------------------------------|
|   1 | GSE101571 |             23 | 200902-GSE101571-full                                |
|   2 | GSE71318  |             48 | 200919-GSE71318-full48                               |
|   3 | GSE133854 |            296 | 200924-GSE133854-all296                              |
|   4 | GSE136447 |            508 | 201109-GSE136447-long508                             |
|   5 | GSE125616 |            640 | 200911-GSE125616-all                                 |
|   6 | GSE44183  |             21 | 201217-GSE44183-earlyhumanlong21                     |
|   7 | GSE72379  |             16 | 201101-GSE72379-full16                               |
|   8 | GSE36552  |            124 | 201104-GSE36552-full124                              |
|   9 | GSE95477  |             20 | 201101-GSE95477-full20                               |
|  10 | GSE65481  |             22 | 201031-GSE65481-full22                               |
|  11 | GSE130289 |            139 | 201031-GSE130289-full139                             |
|  12 | GSE100118 |             92 | 201101-GSE100118-full92                              |
|  13 | GSE49828  |              3 | 201104-GSE49828-RNASeqonly3                          |
|  14 | GSE64417  |             21 | 201218-GSE64417-hESConly21                           |
|  15 | GSE62772  |             18 | 201102-GSE62772-hESC18                               |
|  16 | GSE126488 |             40 | 201103-GSE126488-full40                              |
|  17 | GSE73211  |             30 | 201102-GSE73211-ESC35                                |
|  18 | GSE119324 |             10 | 201104-GSE119324-full10                              |



- Run the following to generate sample metadata files. This script also contains commented codes that put the reads ~r1.fastq.gz~ and ~r2.fastq.gz~ (or ~r.fastq.gz~ for single-ended samples) in the directory ~external/RNA-Seq-with-Run/{name-of-dataset}-{read-length-suffix}/{GSM}/{SRR}/RNA/~. You can modify the path of original fastq files (named with ~YOUR-PATH-with-${srr}-TO~) and uncomment them to deploy the fastq files automatically.
  - NOTE: 1 sample of GSE36552 (GSM922196/SRR491011) has invalid reads (i.e., reads whose sequence length is not equal to the length of quality). We removed the invalid reads from this sample during processing.

#+BEGIN_SRC bash
bash scripts/miscellaneous/generate_sample_metadata_files.sh
#+END_SRC


** 3. Call variants (for RNA editing events identification)

Here we call RNA editing events for each dataset separately. Replace the ~{DATASET_NAME}~ with the dataset name in the table above, and run the command to finish these analyses. 


#+BEGIN_SRC bash
snakemake --snakefile ./pipeline.v3.smk --config threads_indexing=20 threads_trimming=4 threads_aligning=20 threads_merging_bams=1 threads_calling_variants=20 threads_auxiliary_processing=1 threads_auxiliary_processing_parallel=4 default_Xmx='-Xmx60G' --jobs 1 -prk --nolock \
    result/B15_1__get_sample_RNA_editing_sites_v3/{DATASET_NAME}/__merged__/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/finished.step07__apply_complex_filter_1____part09__reformat_data_as_standard_rich_vcf -n
#+END_SRC


*** Example codes for GSE101571


#+BEGIN_SRC bash
snakemake --snakefile ./pipeline.v3.smk --config threads_indexing=20 threads_trimming=4 threads_aligning=20 threads_merging_bams=1 threads_calling_variants=20 threads_auxiliary_processing=1 threads_auxiliary_processing_parallel=4 default_Xmx='-Xmx60G' --jobs 1 -prk --nolock \
    result/B15_1__get_sample_RNA_editing_sites_v3/200902-GSE101571-full/__merged__/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/finished.step07__apply_complex_filter_1____part09__reformat_data_as_standard_rich_vcf -n
#+END_SRC


** 4. Merge variants across datasets

*** 4.1. Specify the total dataset (~210215-sixth-dataset~)

#+BEGIN_SRC sh
cat ./external/DATASET_RNA_EDITING_COLLECTION_NAME_DIRECTORY/{200902-GSE101571-full,200919-GSE71318-full48,200924-GSE133854-all296,201109-GSE136447-long508,200911-GSE125616-all,201217-GSE44183-earlyhumanlong21,201101-GSE72379-full16,201104-GSE36552-full124,201101-GSE95477-full20,201031-GSE65481-full22,201031-GSE130289-full139,201101-GSE100118-full92,201104-GSE49828-RNASeqonly3,201218-GSE64417-hESConly21,201102-GSE62772-hESC18,201103-GSE126488-full40,201102-GSE73211-ESC35,201104-GSE119324-full10} > ./external/DATASET_RNA_EDITING_COLLECTION_NAME_DIRECTORY/210215-sixth-dataset
#+END_SRC

*** 4.2. Merge results across datasets

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part2.smk --config threads_merging_vcfs=36 threads_annotating=18 threads_auxiliary_processing=36 --jobs 1 -prk --nolock \
    result/S16_1__concatenate_RNA_editing_site_from_a_dataset_collection/210215-sixth-dataset/__merged__/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/finished \
    result/S16_3__get_RNA_editing_site_long_table_from_a_dataset_collection/210215-sixth-dataset/__merged__/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/finished -n
#+END_SRC

** 5. Annotate the variants

*** 5.1. Get SnpEff annotations for each variant

Because the raw variants are independent of each other in later filtering steps, here we divide the merged results into different chromosomal bins, process them separately, and merge again the per-bin results into one.

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part2.smk --config threads_merging_vcfs=4 threads_annotating=18 threads_auxiliary_processing=36 --jobs 1 -prk --nolock \ 
    result/S18_1__combine_annotations/210215-sixth-dataset/__merged__/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/snpEff/basic/10000000/{finished.step02__combine_merged_vcf_reformatted_with_snpEff_ANN_split_annotation_dt_filename____patch01__get_full_annotation,finished.step05__combine_merged_variant_only_snpEff_event_summary_dt_filename} -n
#+END_SRC sh

*** 5.2. Check overlap with genomic variants from worldwide cohorts

- Genomic variants spanning chromosomes 1-22, X, and Y:
  - UWashington EVS
  - NCBI ALFA (version 2020.03.04)
  - gnomAD v2.1.1 exomes
  - gnomAD v3.0 genomes

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part2.smk --config threads_annotating=6 threads_auxiliary_processing=6 --jobs 1 -prk --nolock \
    result/S19_1__combine_annotations/210215-sixth-dataset/__merged__/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/bcftools.isec.with.outer.vcf/basic/10000000/{UWashington.EVS,NCBI.ALFA.2020.03.04,gnomAD_v2.1.1_exomes,gnomAD_v3.0_genomes}/collapse_all_and_keep_self_vcf/finished.step06__combine_merged_variant_only_vcf_gz_bcftools_isec_outer_vcf_result_vcf_gz_filename_with_collapse_all_and_keep_self_vcf -n
#+END_SRC

- Genomic variants spanning chromosomes 1-22 and X (i.e., without chromosome Y):
  - 1000Genomes
  - gnomAD v2.1.1 genomes

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part2.smk --config threads_annotating=2 threads_auxiliary_processing=2 --jobs 1 -prk --nolock \
    result/S19_1__combine_annotations/210215-sixth-dataset/__merged__/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/bcftools.isec.with.outer.vcf/basic_without_Y/10000000/{1000Genomes.phased.genotype,gnomAD_v2.1.1_genomes}/collapse_all_and_keep_self_vcf/finished.step06__combine_merged_variant_only_vcf_gz_bcftools_isec_outer_vcf_result_vcf_gz_filename_with_collapse_all_and_keep_self_vcf -n
#+END_SRC

- Combine the results across all cohorts:

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk  --config threads_concatenating_vcfs=36 --jobs 1 -prk --nolock \
    result/S51_1__combine_multiple_population_isec_results/210215-sixth-dataset/finished -n
#+END_SRC

** 6. Generate phenotype table (needed by variant filtering)

*** 6.1. Generate phenotype dataset list

#+BEGIN_SRC sh
mkdir -p external/DATASET_PHENOTYPE_COLLECTION_NAME_DIRECTORY/
echo STUDY,PHENOTYPE_FILENAME > external/DATASET_PHENOTYPE_COLLECTION_NAME_DIRECTORY/201221-fifth-phenotype-collection
for GSE in GSE101571 GSE71318 GSE133854 GSE136447 GSE125616 GSE44183 GSE72379 GSE36552 GSE95477 GSE65481 GSE130289 GSE100118 GSE49828 GSE64417 GSE62772 GSE126488 GSE73211 GSE119324
do
    echo ${GSE},${GSE}.txt >> external/DATASET_PHENOTYPE_COLLECTION_NAME_DIRECTORY/201221-fifth-phenotype-collection
done
#+END_SRC

*** 6.2. Generate phenotype table

NOTE: this ~snakemake~ command costs few CPU cores and memory.

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.smk --jobs 1 -prk --nolock result/S21_1__merge_phenotype_tables/201221-fifth-phenotype-collection/finished -n
#+END_SRC

** 7. Filter for RNA editing events

*** 7.1. Filter against variants overlapping with genomic variants from worldwide cohorts

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk  --config threads_bcftools_isec=36  --jobs 1 -prk --nolock \
    result/S51_2__filter_against_population_variants/210215-sixth-dataset/finished -n
#+END_SRC

*** 7.2. Filter for variants with enough read support

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk  --config threads_bcftools_isec=36  --jobs 1 -prk --nolock \
    result/S51_3__filter_for_variants_with_enough_read_support/210215-sixth-dataset/finished -n
#+END_SRC


*** 7.3. Filter for variants with enough sample support

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk  --config threads_bcftools_isec=36 --jobs 1 -prk --nolock \
    result/S51_4__filter_for_variants_with_enough_sample_support/210215-sixth-dataset/201221-fifth-phenotype-collection/finished -n
#+END_SRC

*** 7.4. Filter for transcribable A-to-G and A-to-G-or-T-to-C variants

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk  --config threads_bcftools_isec=36 --jobs 1 -prk --nolock \
    result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/finished -n
#+END_SRC

*** 7.5. Add snpEff annotation

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk --jobs 1 -prk --nolock \
    result/S51_6__get_snpEff_annotation_subset_of_filtered_result/210215-sixth-dataset/201221-fifth-phenotype-collection/finished -n
#+END_SRC



* Validate the A-to-I identification pipeline on single-cell A375 DNA/RNA-paired sequencing dataset

** 1. Deploy samples

#+BEGIN_SRC bash
bash scripts/miscellaneous/generate_sample_metadata_files_for_A375.sh
#+END_SRC

** 2. Call variants

*** 2.1. A375 DNA

#+BEGIN_SRC bash
snakemake --snakefile ./pipeline.v3.smk --config threads_indexing=20 threads_trimming=1 threads_aligning=20 threads_merging_bams=1 threads_calling_variants=20 threads_auxiliary_processing=1 threads_auxiliary_processing_parallel=4 default_Xmx='-Xmx60G' --jobs 1 -prk --nolock \
    result/B15_1__get_sample_RNA_editing_sites_v3/210203-GSE144296.A375-DNA/__merged__/DNTRSeq-DNA-trimming/hg38.fa/32/bwa-index-default/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/finished.step02__call_variants____part06__really_call_variants -n
#+END_SRC

*** 2.2. A375 RNA

#+BEGIN_SRC bash
snakemake --snakefile ./pipeline.v3.smk --config threads_indexing=20 threads_trimming=1 threads_aligning=20 threads_merging_bams=1 threads_calling_variants=20 threads_auxiliary_processing=1 threads_auxiliary_processing_parallel=4 default_Xmx='-Xmx60G' --jobs 1 -prk --nolock \
    result/B15_1__get_sample_RNA_editing_sites_v3/210203-GSE144296.A375-RNA/__merged__/DNTRSeq-RNA-trimming/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/finished.step07__apply_complex_filter_1____part09__reformat_data_as_standard_rich_vcf -n
#+END_SRC

** 3. Specify the dataset

#+BEGIN_SRC bash

R -e 'library("data.table"); library("magrittr"); geo.dt <- fread("./external/NCBI.SRA.MetaData/GSE144296.txt")[Cell_Line=="A375"][, cell_ID_occurrence:=.N, list(cell_ID)][cell_ID_occurrence==2]; fwrite(geo.dt[LibrarySource=="TRANSCRIPTOMIC", list(TYPE="paired-37-37", DATASET_NAME="210203-GSE144296.A375-RNA-37-37", SAMPLE_NAME=`Sample Name`, INDEXER_PARAMETERS=32)], "external/DATASET_RNA_EDITING_NAME_DIRECTORY/210203-GSE144296.A375-RNA-with-DNA-37-37")'

echo 210203-GSE144296.A375-RNA-with-DNA-37-37 > ./external/DATASET_RNA_EDITING_COLLECTION_NAME_DIRECTORY/210203-GSE144296.A375-RNA-with-DNA-37-37

#+END_SRC

** 4. Merge A375 RNA variants

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part2.smk --config threads_merging_vcfs=36 threads_annotating=18 threads_auxiliary_processing=36 -jobs 1 -prk --nolock \
    result/S16_1__concatenate_RNA_editing_site_from_a_dataset_collection/210203-GSE144296.A375-RNA-with-DNA-37-37/__merged__/DNTRSeq-RNA-trimming/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/finished \
    result/S16_3__get_RNA_editing_site_long_table_from_a_dataset_collection/210203-GSE144296.A375-RNA-with-DNA-37-37/__merged__/DNTRSeq-RNA-trimming/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/finished -n
#+END_SRC

** 5. Annotate A375 RNA varians

*** 5.1. Get SnpEff annotations for each variant


#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part2.smk --config threads_merging_vcfs=4 threads_annotating=18 threads_auxiliary_processing=36  --jobs 1 -prk --nolock \
    result/S18_1__combine_annotations/210203-GSE144296.A375-RNA-with-DNA-37-37/__merged__/DNTRSeq-RNA-trimming/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/snpEff/basic/1000000000/{finished.step02__combine_merged_vcf_reformatted_with_snpEff_ANN_split_annotation_dt_filename____patch01__get_full_annotation,finished.step05__combine_merged_variant_only_snpEff_event_summary_dt_filename} -n
#+END_SRC sh


*** 5.2. Check overlap with genomic variants from worldwide cohorts

- Genomic variants spanning chromosomes 1-22, X, and Y:
  - UWashington EVS
  - NCBI ALFA (version 2020.03.04)
  - gnomAD v2.1.1 exomes
  - gnomAD v3.0 genomes

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part2.smk --config threads_annotating=6 threads_auxiliary_processing=6 --jobs 1 -prk --nolock \
    result/S19_1__combine_annotations/210203-GSE144296.A375-RNA-with-DNA-37-37/__merged__/DNTRSeq-RNA-trimming/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/bcftools.isec.with.outer.vcf/basic/1000000000/{UWashington.EVS,NCBI.ALFA.2020.03.04,gnomAD_v2.1.1_exomes,gnomAD_v3.0_genomes}/collapse_all_and_keep_self_vcf/finished.step06__combine_merged_variant_only_vcf_gz_bcftools_isec_outer_vcf_result_vcf_gz_filename_with_collapse_all_and_keep_self_vcf -n
#+END_SRC


- Genomic variants spanning chromosomes 1-22 and X (i.e., without chromosome Y):
  - 1000Genomes
  - gnomAD v2.1.1 genomes

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part2.smk --config threads_annotating=2 threads_auxiliary_proces sing=2  --jobs 1 -prk --nolock \
    result/S19_1__combine_annotations/210203-GSE144296.A375-RNA-with-DNA-37-37/__merged__/DNTRSeq-RNA-trimming/hg38.fa/32/bwa-index-10.1038_nmeth.2330/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/complex_filter_1/none/bcftools.isec.with.outer.vcf/basic_without_Y/1000000000/{1000Genomes.phased.genotype,gnomAD_v2.1.1_genomes}/collapse_all_and_keep_self_vcf/finished.step06__combine_merged_variant_only_vcf_gz_bcftools_isec_outer_vcf_result_vcf_gz_filename_with_collapse_all_and_keep_self_vcf -n
#+END_SRC


- Combine the results across all cohorts:

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part4.smk  --config threads_concatenating_vcfs=36 --jobs 1 -prk --nolock \
    result/S71_1__combine_multiple_population_isec_results_for_control/210203-GSE144296.A375-RNA-with-DNA-37-37/finished -n
#+END_SRC

** 6. Generate phenotype table (needed by variant filtering)

This step is skipped, because all cells are of the same phenotype (A375 cell line).

** 7. Filter for RNA editing events

*** 7.1. Filter against variants overlapping with genomic variants from worldwide cohorts

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part4.smk  --config threads_bcftools_isec=36 --jobs 1 -prk --nolock \
    result/S71_2__filter_against_population_variants_for_control/210203-GSE144296.A375-RNA-with-DNA-37-37/finished -n
#+END_SRC

*** 7.2. Filter for variants with enough read support

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part4.smk  --config threads_bcftools_isec=36  -prk --nolock \
    result/S71_3__filter_for_variants_with_enough_read_support_for_control/210203-GSE144296.A375-RNA-with-DNA-37-37/finished -n
#+END_SRC


*** 7.3. Filter for variants with enough sample support

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part4.smk  --config threads_bcftools_isec=36 -prk --nolock \
    result/S71_4__filter_for_variants_with_enough_sample_support_for_control/210203-GSE144296.A375-RNA-with-DNA-37-37/210203-GSE144296.A375-RNA-with-DNA-37-37/finished -n
#+END_SRC



*** 7.4. Filter for transcribable A-to-G and A-to-G-or-T-to-C variants

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part4.smk  --config threads_bcftools_isec=36 -prk --nolock \
    result/S71_5__filter_for_A_to_G_sites_for_control/210203-GSE144296.A375-RNA-with-DNA-37-37/210203-GSE144296.A375-RNA-with-DNA-37-37/finished -n
#+END_SRC


* Mark unsequenced sites


** 1. Check variant coverage of each merged bam

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk --config threads_merging_vcfs=1 threads_annotating=1 threads_auxiliary_processing=1 --jobs 1 -prk --nolock \
    result/B52_1__check_variant_converage_of_merged_bam/210215-sixth-dataset/201221-fifth-phenotype-collection/finished -n
#+END_SRC


** 2. Combine all coverage info from bam files

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk --config threads_reduce_and_pigz_compress_tables=36 --jobs 1 -prk --nolock \
    result/S52_2__concatenate_all_variant_coverages_of_merged_bam/210215-sixth-dataset/201221-fifth-phenotype-collection/finished -n
#+END_SRC

** 3. Extract zero and nonzero depth records

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk --config threads_reduce_and_pigz_compress_tables=36 --jobs 1 -prk --nolock \
    result/S52_2__concatenate_all_variant_coverages_of_merged_bam/210215-sixth-dataset/201221-fifth-phenotype-collection/finished.{patch01__extract_zero_depth_records,patch02__extract_nonzero_depth_records} -n
#+END_SRC

** 4. Mark unsequenced sites per samples

- *WARNING*: this step takes about 100GB memory. With 10 threads it finishes within 10-15 minutes.

#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk --config threads_reduce_and_pigz_compress_tables=36 --jobs 1 -prk --nolock \
    result/S52_3__mark_unsequenced_editing_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/finished -n
#+END_SRC


   
* Get expression profile



The expression profile is not needed by identification of RNA editing events, but is needed by analyses related to maternal mRNA clearance.


** 1. Get expression profile for each dataset

Replace the ~{DATASET_NAME}~ with the dataset name in the table above, and run the command to finish these analyses. 


- *WARNING*: steps in expression profiling use the same set of trimmed reads as those used by identification of RNA editing events. Be sure *NOT* to run the steps below when the reads are being trimmed by the step 'Call variants (for RNA editing events identification)'.

#+BEGIN_SRC bash
snakemake --snakefile ./pipeline.v3.smk --config threads_aligning=10 threads_calling_expression=5 threads_auxiliary_processing=1 default_Xmx='-Xmx8G' --jobs 1 -prk --nolock \
    result/BS06_1__get_expression_level/{DATASET_NAME}/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/STAR-expression/__sample_dependent__/STAR-expression/default/stringtie/none/finished.step02__call_expression -n
#+END_SRC

*** Example codes for GSE101571

#+BEGIN_SRC bash
snakemake --snakefile ./pipeline.v3.smk --config threads_aligning=10 threads_calling_expression=5 threads_auxiliary_processing=1 default_Xmx='-Xmx8G' --jobs 1 -prk --nolock \
    result/BS06_1__get_expression_level/200902-GSE101571-full/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/STAR-expression/__sample_dependent__/STAR-expression/default/stringtie/none/finished.step02__call_expression -n
#+END_SRC


** 2. Combine all expression profiles into one

*** 2.1. Specify the total dataset (~210215-sixth-dataset~)

#+BEGIN_SRC bash

cat ./external/DATASET_EXPRESSION_COLLECTION_NAME_DIRECTORY/{200902-GSE101571-full,200919-GSE71318-full48,200924-GSE133854-all296,201109-GSE136447-long508,200911-GSE125616-all,201217-GSE44183-earlyhumanlong21,201101-GSE72379-full16,201104-GSE36552-full124,201101-GSE95477-full20,201031-GSE65481-full22,201031-GSE130289-full139,201101-GSE100118-full92,201104-GSE49828-RNASeqonly3,201218-GSE64417-hESConly21,201102-GSE62772-hESC18,201103-GSE126488-full40,201102-GSE73211-ESC35,201104-GSE119324-full10} > ./external/DATASET_EXPRESSION_COLLECTION_NAME_DIRECTORY/210215-sixth-dataset

#+END_SRC

*** 2.2. Combine all expression profiles into one

#+BEGIN_SRC bash

snakemake --snakefile ./pipeline.v3.part2.smk --config threads_auxiliary_processing=10 --jobs 1 -prk --nolock  \
    result/BS06_1__get_expression_level/210215-sixth-dataset/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/STAR-expression/__sample_dependent__/STAR-expression/default/stringtie/none/finished.step03__get_expression_matrix_by_ballgown -n

#+END_SRC


* Get Total sample count and RE info

** 1. Get total sample count

snakemake --snakefile ./analysis.v1.part1.smk  --jobs 1 -prk --nolock \
    report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.sample.count.for.normal.stages.finished -n
#+END_SRC

** 2. Get RE info

#+BEGIN_SRC sh
snakemake --snakefile ./analysis.v1.part2.smk --jobs 1 -prk --nolock \
    result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/finished -n
#+END_SRC

** 3. Get edit recurrence in GSE133854

#+BEGIN_SRC sh
snakemake --snakefile ./analysis.v1.part2.smk --jobs 1  -prk --nolock \
    result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/GSE133854.all/finished -n
#+END_SRC


* Annotate expression cluster


#+BEGIN_SRC sh
snakemake --snakefile ./pipeline.v3.part3.smk  --jobs 1 -prk --nolock \
    result/S42_1__annotate_embryonic_genes/210215-sixth-dataset/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/STAR-expression/__sample_dependent__/STAR-expression/default/stringtie/none/201221-fifth-phenotype-collection/finished -n
#+END_SRC


* Annotate microRNA binding sites (MBS) and effects of RNA edits on them


** 1. Generate the 3'-UTR region block file

#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --jobs 1 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step01__get_3UTR_blocks/32/gencode.3utr.dt.csv.gz -n
#+END_SRC

** 2. Get 3'-UTR bed file per chromosome 

#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --jobs 1 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step02__get_3UTR_bed_per_chromosome/32/finished -n
#+END_SRC

** 3. Generate MAF fasta per chromosome 

#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --jobs 1 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step03__get_maf_fasta_per_chromosome/32/finished.chr{1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X,Y} -n
#+END_SRC

** 4. Generate inputs for TargetScan (original, unedited 3'-UTR) 

#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --jobs 1 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step04__generate_TargetScan_input_per_chromosome/32/finished.chr{1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X,Y} -n
#+END_SRC

** 5. Run TargetScan on the original 3'-UTR 

#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --jobs 1 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step05__run_TargetScan_per_chromosome/32/finished.chr{1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X,Y} -n
#+END_SRC

** 6. Map each RNA edit to its corresponding 3'-UTR (if any) 

#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --jobs 1 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step06__compute_edit_relative_position_on_3UTR/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/32/gencode.3utr.and.edit.CJ.dt.csv.gz -n
#+END_SRC


** 7. Generate inputs for TargetScan (edited 3'-UTR)

#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --jobs 1 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step07__get_edited_TargetScan_input_per_chromosome/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/32/finished.chr{1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X,Y} -n
#+END_SRC


** 8. Run TargetScan on the edited 3'-UTR

We note that this is not run for chrY, because no 3'-UTR REs were identified on this chromosome.
   
#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --jobs 1 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step08__run_TargetScan_on_edited_per_chromosome/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/32/finished.chr{1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X} -n
#+END_SRC


** 9. Merge TargetScan output for unedited 3'-UTR

#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --cores 10 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step09__concatenate_TargetScan_results_across_all_chromosomes/32/gencode.3utr.all.chromosomes.concatenated.headless.TargetScan.output.gz -n
#+END_SRC


** 10. Merge TargetScan output for edited 3'-UTR 

#+BEGIN_SRC bash
snakemake --snakefile ./analysis.v1.part2.smk  --cores 10 -prk --nolock \
    result/A02_8__get_editing_effect_on_miRNA_binding_sites/step10__concatenate_edited_TargetScan_results_across_all_chromosomes/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/32/edited.gencode.3utr.all.chromosomes.but.chrY.concatenated.headless.TargetScan.output.gz -n
#+END_SRC
  

* Generate figures and tables

See the Zenodo repository (DOI: 10.5281/zenodo.5804921) for the input files needed.

** Figure 1
   
*** Figure 1D

Input:
- Shipped with the git repository:
  - edits identified: ~result/S71_5__filter_for_A_to_G_sites_for_control/210203-GSE144296.A375-RNA-with-DNA-37-37/210203-GSE144296.A375-RNA-with-DNA-37-37/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.dt.txt.gz~
  - sample annotation: ~external/NCBI.SRA.MetaData/GSE144296.txt~
- From Zenodo archive ~pipeline.validation.bcf.files.tar.gz~:
  - bcf files for DNA-Seq: ~result/S15_1__get_sample_RNA_editing_sites_v3/paired-37-37/210203-GSE144296.A375-DNA-37-37/${SAMPLE}/__merged__/DNTRSeq-DNA-trimming/hg38.fa/32/bwa-index-default/DNA/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/alignment.bcf~, where ${SAMPLE} is available from sample annotation
  - bcf files for RNA-Seq: ~result/S15_1__get_sample_RNA_editing_sites_v3/paired-37-37/210203-GSE144296.A375-RNA-37-37/GSM4839055/__merged__/DNTRSeq-RNA-trimming/hg38.fa/32/bwa-index-10.1038_nmeth.2330/32/bwa-aln-samsepe/none/GATK-3.6.0/none/151/common_all/alignment.bcf~, where ${SAMPLE} is available from sample annotation
    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/pipeline.validation/run_internal.R
#+END_SRC

Figure:

- Figure 1D: file:./report.ver2/pipeline.validation/210203-GSE144296.A375/validation.png

*** Figure 2A,2B,2C,2D

Input:
- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
- From Zenodo archive ~editome.files.tar.gz~:
  - All edits identified in each sample: ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.dt.txt.gz~
  - All variants (including edits and other non-edit variants) identified in each sample: ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.with.event.summary.dt.txt.gz~
    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/basic.summary/run_internal.R
#+END_SRC

Figure:

- Figure 2A: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/all.edits.venn.png
- Figure 2B: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/all.edits.count.per.stage.barplot.png
- Figure 2C: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/all.edits.A.to.G.ratio.boxplot.png
- Figure 2D: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/all.edits.Alu.ratio.boxplot.png


*** Figure 2E

Input:
- From [Prerequisites]:
  - Genome fasta file: ~external/contigs/hg38.fa~
  - Genome fasta index: ~external/contigs/hg38.fa~
  - GENCODE transcripts fasta file: ~external/contigs/gencode.v32.transcripts.fa.gz~
- From Zenodo archive ~editome.files.tar.gz~:
  - BED file for all identified edits (editing position only): ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.bed~

    
Run:

#+BEGIN_SRC bash
snakemake --cores 1 --snakefile ./analysis.v1.part1.smk -prk result/A01_5__plot_motif/210215-sixth-dataset/201221-fifth-phenotype-collection/finished
#+END_SRC

Figure:

- Figure 2E: file:./result/A01_5__plot_motif/201218-fifth-dataset/201221-fifth-phenotype-collection/temp_result_directory/temp.b3.extended.sequence.lines.tsl.plot.png


*** Figure 2F


Input:
- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
- From Zenodo archive ~editome.files.tar.gz~:
  - All edits identified in each sample: ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.dt.txt.gz~

Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/basic.summary/run_internal_plot_example.R
#+END_SRC

Figure:

- Figure 2F: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/BLCAP.example.png


*** Figure 3B,3C,3D,3E, and Supplementary Figure 1

Input:
- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.dt.txt.gz~
  - Occurrence of RE-matching edits in each normal sample: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.site.recurrence.comparison.CJ.dt.txt.gz~
  - Observed edits identified in each normal sample, on valid genes only: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.observed.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~

    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/RE/run_internal.R
#+END_SRC

Figure:

- Figure 3B: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE/RE.count.per.stage.png
- Figure 3C: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE/all.edit.types.piechart.png
- Figure 3D: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE/3UTR.ratio.barplot.png
- Supplementary Figure 1: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE/3UTR.ratio.barplot.for.observed.edits.png
- Figure 3E: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE/RE.sankey.png



*** Figure 4B,4C

Input:
- Shipped with the git repository:
  - Total sample count for each normal stage: ~./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.sample.count.for.normal.stages.dt.csv~
  - Stage description: ~./manuscript/table_for_processing.xlsx~
- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~


    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/RE.gene/run_internal.R
#+END_SRC

Figure:

- Figure 4B: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.gene/RE.gene.count.per.stage.png
- Figure 4C: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.gene/RE.gene.sankey.png



*** Figure 4D

Input:
- Shipped with the git repository:
  - Table for RE-targeted genes: ~./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.gene/RE.targeted.genes.dt.csv.gz~ (also generated by ~./scripts.for.report.ver2/RE.gene/run_internal.R~)
  - Stage description: ~./manuscript/table_for_processing.xlsx~

Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/RE.gene.GO/run_internal.R
#+END_SRC

Figure:

- Figure 4D: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.gene/RE.gene.GO.tileplot.png



*** Supplementary Table 3,4, Figure 5A,5B

Input:
- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
- From Zenodo archive ~editome.files.tar.gz~:
  - All edits identified in each sample: ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.dt.txt.gz~
- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~

    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/phenotypic.check/run_internal_disease_lost_RE_check.R
#+END_SRC

Table:

- Supplementary Table 3: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/disease.or.old.mother.embryo.lost.RE.dt.csv.gz
- Supplementary Table 4: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/disease.and.old.mother.embryos.lost.RE.all.enrichGO.results.combined.dt.csv.gz

Figure:

- Figure 5A: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/disease.or.old.mother.embryo.lost.RE.count.barplot.png
- Figure 5B: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/disease.or.old.mother.embryo.lost.RE.gene.GO.gene.count.ge.4.barplot.png



*** Figure 5C


Input:
- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
- From Zenodo archive ~editome.files.tar.gz~:
  - All edits identified in each sample: ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.dt.txt.gz~
- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~

Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/phenotypic.check/run_internal_overall_RE_count_check_2nd.R
#+END_SRC

Figure:

- Figure 5C: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/normal.valid.gene.only.RE.count.histogram.three.datasets.png


*** Figure 5E, Supplementary Figure 2A,2B

Input:
- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
  - maternal gene expression and annotation: ~result/S42_1__annotate_embryonic_genes/210215-sixth-dataset/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/STAR-expression/__sample_dependent__/STAR-expression/default/stringtie/none/201221-fifth-phenotype-collection/combined.gexpr.FPKM.pc.only.melt.with.phenotype.normal.sample.only.median.annotated.dt.txt.gz~
- From Zenodo archive ~editome.files.tar.gz~:
  - All edits identified in each sample: ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.dt.txt.gz~
- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~

  
    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/phenotypic.check/run_internal_GSE95477.R
#+END_SRC

Figure:

- Supplementary Figure 2A: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/RE.count.boxplot.GSE95477.png
- Supplementary Figure 2B: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/GSE95477.oocyte.hclust.png
- Figure 5E: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/RE.count.boxplot.GSE95477.without.GSM2514781.and.GSM2514773.png



*** Supplementary Figure 3A

Input:
- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
  - maternal gene expression and annotation: ~result/S42_1__annotate_embryonic_genes/210215-sixth-dataset/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/STAR-expression/__sample_dependent__/STAR-expression/default/stringtie/none/201221-fifth-phenotype-collection/combined.gexpr.FPKM.pc.only.melt.with.phenotype.normal.sample.only.median.annotated.dt.txt.gz~
- From Zenodo archive ~editome.files.tar.gz~:
  - All edits identified in each sample: ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.dt.txt.gz~
- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~

    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/phenotypic.check/run_internal_GSE101571.R
#+END_SRC

Figure:

- Supplementary Figure 3A: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/RE.count.boxplot.GSE101571.png

*** Supplementary Figure 3B

Input:
- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
  - maternal gene expression and annotation: ~result/S42_1__annotate_embryonic_genes/210215-sixth-dataset/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/STAR-expression/__sample_dependent__/STAR-expression/default/stringtie/none/201221-fifth-phenotype-collection/combined.gexpr.FPKM.pc.only.melt.with.phenotype.normal.sample.only.median.annotated.dt.txt.gz~
- From Zenodo archive ~editome.files.tar.gz~:
  - All edits identified in each sample: ~result/S51_5__filter_for_A_to_G_sites/210215-sixth-dataset/201221-fifth-phenotype-collection/merged.long.disjoint.with.population.without.potential.polymorphism.with.enough.read.support.with.phenotype.sequenced.samples.only.with.enough.sample.support.A.to.G.only.dt.txt.gz~
- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~

    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/phenotypic.check/run_internal_GSE133854.R
#+END_SRC

Figure:

- Supplementary Figure 3B: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.samples/RE.count.boxplot.GSE133854.png




*** Figure 6B,6C,6D, Supplementary Figure 4

Input: 

- Prerequisites:
  - Reference GTF file: ~external/reference.gene.annotation/GENCODE.annotation/32/gencode.annotation.gtf~
- From Zenodo archive ~microRNA.files.tar.gz~:
  - TargetScan output for unedited transcripts: ~result/A02_8__get_editing_effect_on_miRNA_binding_sites/step09__concatenate_TargetScan_results_across_all_chromosomes/32/gencode.3utr.all.chromosomes.concatenated.headless.TargetScan.output.gz~.
  - TargetScan output for edited transcripts: ~result/A02_8__get_editing_effect_on_miRNA_binding_sites/step10__concatenate_edited_TargetScan_results_across_all_chromosomes/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/32/edited.gencode.3utr.all.chromosomes.but.chrY.concatenated.headless.TargetScan.output.gz~


First run the preparation script:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/miRNA/run_internal_prepare_ts.R
#+END_SRC

Then run the following:

**** Figure 6B

Input:

- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
- Prerequisites:
  - Reference GTF file: ~external/reference.gene.annotation/GENCODE.annotation/32/gencode.annotation.gtf~
- From the prepartion step above:
  - MBS comparison between unedited and edited: ~./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/miRNA/edited.ts.human.compared.with.original.dt.csv.gz~ (also shipped with git repository)
- From Zenodo archive ~RE.files.tar.gz~:
  - Observed edits identified in each normal sample, on valid genes only: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.observed.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~

     
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/miRNA/run_internal_piechart.R
#+END_SRC

- Figure 6B: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.miRNA/3UTR.types.piechart.png

**** Figure 6C,6D, Supplementary Figure 4

Input:

- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
  - maternal gene expression and annotation: ~result/S42_1__annotate_embryonic_genes/210215-sixth-dataset/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/STAR-expression/__sample_dependent__/STAR-expression/default/stringtie/none/201221-fifth-phenotype-collection/combined.gexpr.FPKM.pc.only.melt.with.phenotype.normal.sample.only.median.annotated.dt.txt.gz~
- Prerequisites:
  - Reference GTF file: ~external/reference.gene.annotation/GENCODE.annotation/32/gencode.annotation.gtf~
- From the prepartion step above:
  - MBS comparison between unedited and edited, summarized at gene and edit level: ~./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.miRNA/edited.ts.human.compared.with.original.annotated.summary.gene.and.edit.level.dt.csv.gz~ (also shipped with the git repository)
- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~


     
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/miRNA/run_internal_MBS_count.R
#+END_SRC

- Figure 6C: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.miRNA/RE.MBS.gain.and.loss.count.boxplot.png
- Figure 6D: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.miRNA/RE.MBS.gain.count.on.MTC.or.nonMTC.maternal.genes.barplot.png
- Supplementary Figure 4: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.miRNA/RE.MBS.net.change.count.on.MTC.or.nonMTC.maternal.genes.barplot.png
  


*** Track file for visualizing Figure 6E on UCSC Genome Browser


Input:

- Shipped with the git repository:
  - maternal gene expression and annotation: ~result/S42_1__annotate_embryonic_genes/210215-sixth-dataset/auto-detect-and-cut-adapter-by-trim-galore-and-select-reads-with-base-quality-no-smaller-than-25-by-fastp/hg38.fa/32/STAR-expression/__sample_dependent__/STAR-expression/default/stringtie/none/201221-fifth-phenotype-collection/combined.gexpr.FPKM.pc.only.melt.with.phenotype.normal.sample.only.median.annotated.dt.txt.gz~
- Prerequisites:
  - Reference GTF file: ~external/reference.gene.annotation/GENCODE.annotation/32/gencode.annotation.gtf~
- From the prepartion step above:
  - MBS comparison between unedited and edited, summarized at gene and edit level: ~./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.miRNA/edited.ts.human.compared.with.original.annotated.summary.gene.and.edit.level.dt.csv.gz~ (also shipped with the git repository)

    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/miRNA/run_internal_MBS_case_check.R
#+END_SRC

- Track file: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.miRNA/case.study.bed

Note that to visualize the color, one needs to put ~itemRgb="On"~ in the track configuration.

    
*** Supplementary Table 5

Input:

- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.dt.txt.gz~
  - Edit recurrence in each GSE133854 sample: ~./result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/GSE133854.all/subset.site.recurrence.comparison.CJ.dt.txt.gz~


    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/UPD.tables/run_internal_RE.R
#+END_SRC

Table:

- Supplementary Table 5: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/GSE133854.all/normal.RE.lost.in.UPD.dt.csv.gz


*** Supplementary Table 6


Input:

- From Zenodo archive ~RE.files.tar.gz~:
  - Edits in normal samples: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.dt.txt.gz~ 
  - SnpEff annotation for observed edits identified in normal samples, on valid genes only: ~./result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/snpEff.annotation.for.subset.observed.edits.dt.txt.gz~

    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/recoding.edits/run_internal_recoding.R
#+END_SRC

Table:

- Supplementary Table 6: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/normal.edits.subset.recoding.only.dt.csv.gz


*** Supplementary Table 7


Input:
- Shipped with the git repository:
  - Stage description: ~./manuscript/table_for_processing.xlsx~
  - Total sample count for each normal stage: ~./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/total.sample.count.for.normal.stages.dt.csv~
- From Zenodo archive ~RE.files.tar.gz~:
  - REs identified in each normal sample, on valid genes only, with snpEff annotation: ~result/A02_4__check_fine_recurrence_profile_for_a_subset_of_samples/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/subset.recurrent.edits.only.with.snpEff.annotation.on.valid.genes.only.dt.txt.gz~

    
Run:

#+BEGIN_SRC bash
Rscript ./scripts.for.report.ver2/postimplantation.RE.gene/run_internal.R
#+END_SRC

Table:

- Supplementary Table 7: file:./report.ver2/210215-sixth-dataset/201221-fifth-phenotype-collection/all.normal.samples/RE.gene/postimplantation.RE.genes.and.their.RE.dt.csv.gz
